### PR TITLE
Add replication (to Google Drive)

### PR DESCRIPTION
**Pull request by arhamahmed**
Google Drive is added as a new storage sink. User's will need to [enable](
https://console.developers.google.com) the Drive API on their Google account and obtain an authorization token used for Google's OAuth2 flow. Streamed writes to Drive are a work in progress.

### CODE REVIEW

File: cmd/storage-server/run.go [hunk 1]
```diff
@@ -76,22 +78,37 @@ func Start(logger *log.Logger, port int, storePath string) (wait func(), shutdow
...
+
+	gdriveStore := &gdriveServer{}
+	err = gdriveStore.Setup(logger)
+	if err != nil {
+		logger.Errorf("Unable to setup Google Drive: %v", err)
+	} else {
+		storageSinks = append(storageSinks, gdriveStore)
+	}
+
+	if !storeRemotely {
```
Comment (Dendrimer) [inline cmd/storage-server/run.go:91]: We can't store remotely and store locally?

File: cmd/storage-server/run.go [hunk 2]
```diff
@@ -76,22 +78,37 @@ func Start(logger *log.Logger, port int, storePath string) (wait func(), shutdow
...
-		InverseTransform: func(pk *diskv.PathKey) string {
-			return strings.Join(pk.Path, "/") + pk.FileName
-		},
-	})
-	storageServer := newStorageServer(logger, store)
+	var storageSinks []Store
+
+	gdriveStore := &gdriveServer{}
+	err = gdriveStore.Setup(logger)
+	if err != nil {
```
Comment (Dendrimer) [inline cmd/storage-server/run.go:85]: It's kind of questionable to want to set up google drive by default, perhaps we want to make this an option specified at startup time from the command line

**Review by Dendrimer — COMMENTED** [review]
We should probably think about how we can rollback destructive operations 
or have some mechanism of getting into consistent state across stores assuming
a delete operation fails...

Also some workflow issues to think about (like defaulting to starting Google drive) to think about but code looks good.

File: cmd/storage-server/run.go (hunk 1 above)
Comment (arhamahmed) [reply cmd/storage-server/run.go:91]: We can, it's default behaviour. The flag is meant for the case where local storage space is low so store everything remotely.

File: cmd/storage-server/run.go (hunk 2 above)
Comment (irfansharif) [reply cmd/storage-server/run.go:85]: Definitely not a default option.

**Comment by irfansharif** [comment]
RIP Kura (2018 - 2019).

**Pull request closed without merging**